### PR TITLE
cache handling in volume - address #6285

### LINKED ISSUE #6285: [FEATURE] Generic cache management for`module_set_large_config()`
**Is your feature request related to a problem? Please describe.**
A problem with cache management has recently been fixed by #6274 for the volume module. However, that fix has to be made generic, not volume-specific.

**Describe the solution you'd like**
Cache invalidation for module configuration has to be dome in module core, not in individual module drivers.

**Additional context**
See the aforementioned PR and particularly comment https://github.com/thesofproject/sof/pull/6274#issuecomment-1241610167 in it.

### PR DESCRIPTION
**Pull request by lyakh**
clarify IPC-version specific processing paths and remove a redundant cache invalidation. This should resolve #6285

### CODE REVIEW

**Review by lgirdwood — APPROVED** [review]
LGTM.

**Comment by lgirdwood** [comment]
SOFCI TEST

**Comment by keqiaozhang** [comment]
SOFCI TEST

File: src/include/sof/audio/module_adapter/module/generic.h [hunk 1]
```diff
@@ -21,10 +21,6 @@
 #include <sof/audio/dp_queue.h>
 #include "module_interface.h"
 
-#if CONFIG_INTEL_MODULES
-#include "modules.h"
-#endif
-
```
Comment (kv2019i) [inline src/include/sof/audio/module_adapter/module/generic.h:27]: How is this related to the PR?

File: src/include/sof/audio/module_adapter/module/modules.h [hunk 2]
```diff
@@ -44,22 +45,27 @@ struct comp_dev *modules_shim_new(const struct comp_driver *drv,
...
-	(comp_dynamic_module)->ops.copy = module_adapter_copy; \
-	(comp_dynamic_module)->ops.cmd = module_adapter_cmd; \
-	(comp_dynamic_module)->ops.trigger = module_adapter_trigger; \
-	(comp_dynamic_module)->ops.reset = module_adapter_reset; \
-	(comp_dynamic_module)->ops.free = module_adapter_free; \
-	(comp_dynamic_module)->ops.set_large_config = module_set_large_config;\
-	(comp_dynamic_module)->ops.get_large_config = module_get_large_config;\
-	(comp_dynamic_module)->ops.get_attribute = module_adapter_get_attribute; \
-} while (0)
+static inline void declare_dynamic_module_adapter(struct comp_driver *drv,
```
Comment (kv2019i) [inline src/include/sof/audio/module_adapter/module/modules.h:48]: Why this change? Looks good itself, but I'm failing to make the conection to the git commit description.

**Review by kv2019i — COMMENTED** [review]
Code looks good, but the PR title is a bit misleading. This seems to do a bigger cleanup than fix cache handling in volume. And some bits in first commit go beyond removing ".cmd()". Is there rationale why to do all of this in single commit?

File: src/include/sof/audio/module_adapter/module/modules.h (hunk 2 above)
Comment (lyakh) [reply src/include/sof/audio/module_adapter/module/modules.h:48]: this makes excluding `.cmd`  for IPC4 easy

**Comment by lyakh** [comment]
> Code looks good, but the PR title is a bit misleading. This seems to do a bigger cleanup than fix cache handling in volume. And some bits in first commit go beyond removing ".cmd()". Is there rationale why to do all of this in single commit?

@kv2019i it would be possible split it up a little bit, e.g. make the conversion of a macro to an inline function a separate patch but the whole reason why I did that was to remove `.cmd()` so it really all got pulled in by that single change

**Review by kv2019i — APPROVED** [review]
Thanks @lyakh for clarifications. Then let's merge in one go. 

Still needs a clean mandatory CI run, then ready to be merged.
